### ISSUE TITLE
Cannot specify commit message

### ISSUE THREAD

**Issue by nagytech** [microsoft/vscode-vsce#364]
I can see that `vsce publish` discretely uses `npm version` to apply versioning to `package.json` which also adds a commit  to the local repository.  The default commit message turns out to be '[major].[minor].[patch]', and there is currently no way to override this.

When using `vsce publish` as a part of a CI/CD pipeline, I need to be able to specify a commit message that  includes an ignore flag so that builds are not re-triggered by the build agent pushing changes.

**Comment by nagytech** [comment]
Related to #9

**Comment by joaomoreno** [comment]
@nagytech Why not running `npm version -m MESSAGE` yourself and then simply `vsce publish`?

**Comment by nagytech** [comment]
Because `vsce publish` is already calling `npm version` there is an expectation that the versioning should be handled internally.

**Comment by raix** [comment]
@nagytech I don't follow the title of this issue completely - but I'll share a couple of options, depending on your setup:

---

I use semantic release with the vsce plugin to automate the CI for the perl debug extension on github, takes some practice to get into - but on the other hand we get release notes/semantic versioning for free and contributors don't have to wait for me to publish.

---

It's also possible to setup a multi-branch build pipeline with a counter for the version and only publish when committing to master (additional add pre releases eg. on feature branches) or setup a release pipeline for promotion of versions manually. _(Azure Pipelines works well in this case but jenkins also supports this - and I would be surprised if travis is lagging behind)_

_(in these cases the version is tracked by the build system not package.json eg. `vsce publish ${BuildNumber}` - so no additional commits - optional tagging of the version on successful builds)_

**Comment by nagytech** [comment]
The build process I'm following is as follows:

```
vsce publish minor -p $PAT   
git push origin master
```

As noted before, `vsce publish [version]` will bump the version in `package.json` _and_ add a commit to the local git repository.  The problem here is that there are no options to control the format of the commit message.  This causes cyclical builds which I'm trying to avoid.

So far, all of the options I've explored rely on external tools, processes or adding additional steps to my build.  What I really want is to be able to run the two existing build commands (with a custom commit message) and walk away without worrying about CI/CD imploding.

The changes I've already submitted help me accomplish this and are done in a way that is as unobtrusive as possible.

Hopefully that's not too much to ask.

**Comment by joaomoreno** [comment]
VSCE only bumps the version because you ask it to by providing `minor`. Try this instead:

```
npm version minor -m MESSAGE
vsce publish -p $PAT
git push origin master
```

**Comment by nagytech** [comment]
I appreciate the help, but I'm well aware of how I can do that.  Obviously I would have needed to understand how `npm version` works in order to make the changes in this pull request: https://github.com/microsoft/vscode-vsce/pull/365

**Comment by joaomoreno** [comment]
Fixed by https://github.com/microsoft/vscode-vsce/pull/365

**Comment by joaomoreno** [comment]
Published 1.64.0.
